crates/oxc_query/src/edges.rs

```diff
@@ -0,0 +1,2168 @@
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    edge_name: &str,
+    _parameters: &EdgeParameters,
+    resolve_info: &ResolveEdgeInfo,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+    match edge_name {
+        "span" => jsxattribute::span(contexts, resolve_info),
+        "value_as_expression" => jsxattribute::value_as_expression(contexts, resolve_info),
+        "value_as_url" => jsxattribute::value_as_url(contexts, resolve_info),
+        _ => {
+            unreachable!(
+                "attempted to resolve unexpected edge '{edge_name}' on type 'JSXAttribute'"
+            )
+        }
+    }
+}
+
+mod jsxattribute {
+    use oxc_ast::ast::JSXAttributeValue;
+    use trustfall::provider::{
+        resolve_neighbors_with, ContextIterator, ContextOutcomeIterator, ResolveEdgeInfo,
+        VertexIterator,
+    };
+
+    use super::{super::vertex::Vertex, get_span};
+
+    pub(super) fn span<'a, 'b: 'a>(
+        contexts: ContextIterator<'a, Vertex<'b>>,
+        _resolve_info: &ResolveEdgeInfo,
+    ) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+        get_span(contexts)
+    }
+
+    pub(super) fn value_as_expression<'a, 'b: 'a>(
+        contexts: ContextIterator<'a, Vertex<'b>>,
+        _resolve_info: &ResolveEdgeInfo,
+    ) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+        resolve_neighbors_with(contexts, |v| {
+            let attr = v.as_jsx_attribute().unwrap_or_else(|| {
+                panic!("expected to have a jsxattribute vertex, instead have: {v:#?}")
+            });
+            let Some(attr_value) = &attr.value else { return Box::new(std::iter::empty()) };
+            Box::new(
+                std::iter::once(match attr_value {
+                    JSXAttributeValue::ExpressionContainer(expr) => match &expr.expression {
+                        oxc_ast::ast::JSXExpression::Expression(expr) => Some(Vertex::from(expr)),
+                        oxc_ast::ast::JSXExpression::EmptyExpression(_) => None,
+                    },
+                    JSXAttributeValue::Fragment(_)
+                    | JSXAttributeValue::StringLiteral(_)
+                    | JSXAttributeValue::Element(_) => None,
+                })
+                .flatten(),
+            )
+        })
+    }
+
+    pub(super) fn value_as_url<'a, 'b: 'a>(
+        contexts: ContextIterator<'a, Vertex<'b>>,
+        _resolve_info: &ResolveEdgeInfo,
+    ) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+        resolve_neighbors_with(contexts, |v| {
+            #[allow(clippy::option_if_let_else)]
+            if let Some(url) = Vertex::make_url(v.as_jsx_attribute().unwrap_or_else(|| {
+                panic!("expected to have a jsxattribute vertex, instead have: {v:#?}")
+            })) {
+                Box::new(std::iter::once(url))
+            } else {
+                Box::new(std::iter::empty())
+            }
```

Comment:
use the option as an iterator directly here as well

Reply:
I did on the rest of the if let Some's